### ISSUE TITLE
how to use my own data(not image) to forcast.

### ISSUE THREAD

**Issue by sunjia2015** [BVLC/caffe#3618]
I have trained my network with my own data, which is not a image but a vector(20×128×128).
Now I want to use the trained network to forecast.
However, when I try the code,
I got the same output no matter what the input vector is. 
I will attach the file that I have used .
Any advice would be appreciated. 

`import cv2
import numpy as np
import os
import caffe
import lmdb
import leveldb
from caffe.proto import caffe_pb2

caffe_root = '../' 
import sys
sys.path.insert(0, caffe_root + 'python')

def read_txt(file_path):
    X = np.empty((20,128,128),dtype=np.float)
    if os.path.exists(file_path) == False:
       return X
    f = open(file_path)
    lines2 = f.readlines()
    f.close()        
    count = -1
    for line in lines2:
        count = count + 1
	i = count % 130
        k = count//130
	if i <= 127 and k<20:
	    words = line.split(',')
	    for j in range(0,128):
	    X[k,i,j]=float(words[j])/500000
        #print '********************',X.shape
        return X

MODEL_FILE = '***************/try_bn_deploy.prototxt'
PRETRAINED = '***************/bn__iter_20000.caffemodel'
base_path = '****************/database/fx/txt/'

f = open('*****************/val.txt')
line = f.readline()
caffe.set_mode_cpu()
net = caffe.Net(MODEL_FILE, PRETRAINED,caffe.TEST)

count = 0
correct = 0
while line:
    count += 1    
    # load feature vector
    pair = line.strip().split()
    image_path = pair[0]
    label = pair[1]
    X=np.empty((1,20,128,128),dtype=np.float)
    X[0,:,:,:] = read_txt(base_path + image_path)
    net.blobs['data'].data[...] = X
    # forcast
    out = net.forward() 
    plabel = int(out['prob'][0].argmax())
    print count, label, plabel
    line = f.readline()
f.close`

This is my deploy file

`name: "Net"
input: "data"
input_shape {
  dim: 1
  dim: 20
  dim: 128
  dim: 128
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  convolution_param {
    num_output: 96
    kernel_size: 7
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_term: false
  }
}
layer {
  name: "conv1_bn"
  type: "BatchNorm"
  bottom: "conv1"
  top: "conv1_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "sigmoid1"
  type: "Sigmoid"
  bottom: "conv1_bn"
  top: "sigmoid1"
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "sigmoid1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "pool1_bn"
  type: "BatchNorm"
  bottom: "pool1"
  top: "pool1_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1_bn"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_term: false
  }
}
layer {
  name: "conv2_bn"
  type: "BatchNorm"
  bottom: "conv2"
  top: "conv2_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "sigmoid2"
  type: "Sigmoid"
  bottom: "conv2_bn"
  top: "sigmoid2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "sigmoid2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "pool2_bn"
  type: "BatchNorm"
  bottom: "pool2"
  top: "pool2_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "pool2_bn"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_term: false
  }
}
layer {
  name: "conv3_bn"
  type: "BatchNorm"
  bottom: "conv3"
  top: "conv3_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer{
  name: "sigmoid3"
  type: "Sigmoid"
  bottom: "conv3_bn"
  top: "sigmoid3"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "sigmoid3"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "pool5_bn"
  type: "BatchNorm"
  bottom: "pool5"
  top: "pool5_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5_bn"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1024
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "fc6_bn"
  type: "BatchNorm"
  bottom: "fc6"
  top: "fc6_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "sigmoid6"
  type: "Sigmoid"
  bottom: "fc6_bn"
  top: "sigmoid6"
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "sigmoid6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1024
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "fc7_bn"
  type: "BatchNorm"
  bottom: "fc7"
  top: "fc7_bn"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  batch_norm_param{
    use_global_stats: false
  }
}
layer {
  name: "sigmoid7"
  type: "Sigmoid" 
  bottom: "fc7_bn"
  top: "sigmoid7"
}

layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "sigmoid7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  inner_product_param {
    num_output:14 
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "prob"
  type: "Softmax"
  bottom: "fc8"
  top: "prob"
}`

**Comment by seanbell** [comment]
From [CONTRIBUTING.md](https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md):
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
